### ISSUE TITLE
Creating a new branch automatically hides (or filters) the new branch

### ISSUE THREAD

**Issue by bdgeyter** [fork-dev/Tracker#185]
After the commit filter was introduced, creating a new branch results in that branch being marked as filtered out. Commits to that branch are not visible in All Commits, until you press the filter icon next to the hidden branch. Also the branch name is faded out.
<img width="213" alt="screen shot 2017-12-21 at 10 09 18" src="https://user-images.githubusercontent.com/4182170/34248602-0bfc15c2-e637-11e7-8d6d-3aac8cb89e04.png">

**Comment by DanPristupov** [comment]
@bdgeyter hi. Are you using the latest 1.0.61 version?

**Comment by bdgeyter** [comment]
Yep

**Comment by DanPristupov** [comment]
I just checked the source and found the suspicious part of code. I'll fix the bug after the holidays.
Merry Christmas!

**Comment by bdgeyter** [comment]
Great! Enjoy the holidays!

**Comment by DanPristupov** [comment]
@bdgeyter sorry, the suspicions were wrong and I could not reproduce the problem. Do you still face it? Does it happen every time you create a branch?

**Comment by bdgeyter** [comment]
I’ve seen it recently still but when I tried to reproduce I didn’t manage so it definitely doesn’t happen every time. I’ll try and pinpoint the issue if it appears again. Feel free to close for now.

**Comment by bdgeyter** [comment]
<img width="340" alt="screen shot 2018-01-19 at 10 27 29" src="https://user-images.githubusercontent.com/4182170/35143955-5f4511f6-fd03-11e7-90a7-cc15b5d41340.png">
just happened again

**Comment by DanPristupov** [comment]
May this is caused by subfolders. I'll try to reproduce this with nested structure.

**Comment by DanPristupov** [comment]
@bdgeyter I spent about an hour reproducing this bug, still no luck.

Do you remember if you had some filters already _before_ you created a new branch? Is it possible that the new branch became hidden because filter had been set on two other branches?

**Comment by bdgeyter** [comment]
No, that wasn't the case for my local branches, but some more info @DanPristupov: it seems like lots of branches from my origin are filtered (greyed out), even though there's no more active filter visible in the list. Is there any way this could get screwed up if you have a filter set and then do a fetch of new remote branches? Or is there an easy way to show which branches have an active filter?

**Comment by bdgeyter** [comment]
Seeing the behaviour of filters more in depth now on the test repo I made, I do think it could be helpful to add a feature to remove all filters. I guess the behaviour could be explained if I have a single filter set on a branch that is not immediately visible in the tree (and eg. deeply nested). My solution has been to click the filter icon of the newly created branch that looked like it was filtered out (= greyed out), but I realise now this adds an extra filter for this branch to the list of filters, only making the problem worse.

Long story short, it seems my issue is non-issue, but it might needs some UI to deselect all filters, or indicate how many filters are active...

**Comment by bdgeyter** [comment]
Actually, I realise now this behaviour is even visible in the screenshot I sent above, where two filters are selected...

I'm not completely sure, but I'd expect `Create New Branch` to clear all active filters, or was there a specific use case for this?
